### PR TITLE
Zabbix user and password

### PR DESCRIPTION
**Pull request by fegalo**
##### SUMMARY
There wasn't any option to include user and password for Zabbix API
(They are different fields to _Basic Auth_ username and password)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
grafana_datasource

### CODE REVIEW

File: tests/integration/targets/grafana_datasource/tasks/zabbix.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    - result.datasource.name == 'datasource-zabbix'
+    - result.datasource.orgId == 1
+    - result.datasource.password == ''
+    - result.datasource.type == 'alexanderzobnin-zabbix-datasource'
+    - result.datasource.url == 'https://zabbix.example.com'
+    - result.datasource.user == ''
+    - not result.datasource.withCredentials
+    - "result.msg == 'Datasource datasource-zabbix updated'"
+
+- name: Delete elasticsearch datasource
```
Comment (rrey) [inline tests/integration/targets/grafana_datasource/tasks/zabbix.yml:90]:
```suggestion
- name: Delete zabbix datasource
```

File: tests/integration/targets/grafana_datasource/tasks/zabbix.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    zabbix_user: grafana
+    zabbix_password: '******'
+    tls_ca_cert: /etc/ssl/certs/ca.pem
+    state: absent
+
+- assert:
+    that:
+    - result.changed
+
+- name: Delete elasticsearch datasource (idempotency)
```
Comment (rrey) [inline tests/integration/targets/grafana_datasource/tasks/zabbix.yml:109]:
```suggestion
- name: Delete zabbix datasource (idempotency)
```

File: tests/integration/targets/grafana_datasource/tasks/zabbix.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    that:
+    - result.changed
+    - not result.datasource.isDefault
+    - result.datasource.jsonData.username == 'grafana'
+    - result.datasource.name == 'datasource-zabbix'
+    - result.datasource.orgId == 1
+    - result.datasource.password == ''
+    - result.datasource.type == 'alexanderzobnin-zabbix-datasource'
+    - result.datasource.url == 'https://zabbix.example.com'
+    - result.datasource.user == ''
```
Comment (rrey) [inline tests/integration/targets/grafana_datasource/tasks/zabbix.yml:86]: Is that expected or a bug ? User should not be empty right ?

Comment (fegalo) [reply tests/integration/targets/grafana_datasource/tasks/zabbix.yml:86]: I am not sure, I copied the elastic test.

_result.datasource.jsonData.username_ has to match with _zabbix_user_. We don't need to check _result.datasource.user_ and _result.datasource.password_ for this example, maybe we can remove them.

Comment (rrey) [reply tests/integration/targets/grafana_datasource/tasks/zabbix.yml:86]: ok, it's not new then.

**Comment by rrey** [comment]
Thanks for contributing !

**Pull request merged**